test/testXliffMerge.js

```diff
@@ -0,0 +1,171 @@
+/*
+ * testXliffMerge.js - test the merge of Xliff object.
+ *
+ * Copyright © 2020 JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+if (!Xliff) {
+    var Xliff = require("../lib/Xliff.js");
+}
+
+module.exports.xliffmerge = {
+    testXliffMergeenUS: function(test) {
+        test.expect(2);
+        
+        var fs = require("fs");            
+        var target = new Xliff({
+            version: 2
+        });
+
+        test.ok(target);
+
+        var files = [
+            "testfiles/xliff20/app1/en-US.xliff",
+            "testfiles/xliff20/app2/en-US.xliff",
+        ];
+
+        files.forEach(function(file) {
+            if (fs.existsSync(file)){
```

Comment:
This replicates the code in loctool.js, which proves it works but does not test it directly. If someone changes the loctool.js code, then this code will not be testing the same thing anymore. In order to test it properly, you should extract the merge (and the split) code into a new file that this unit test file can load. Then, both loctool.js and these unit tests can call that code. That way, we are guaranteed that we are testing the same code that people will use.